support hasura naming conventions

joshuadutton opened the pull request:
- All tests pass when `naming_convetion: hasura-default`

joshuadutton commented:
The latest commit is to preemptively help prepare for when Nhost uses [Hasura v2.21.0](https://github.com/hasura/graphql-engine/releases/tag/v2.21.0) or greater since more of the `graphql-default` naming conventions were changed in that release. 

The commit uses the alias `affectedRows: affected_rows`.

szilarddoro commented on docker-compose.yaml:
```diff
@@ -22,6 +22,8 @@ services:
       HASURA_GRAPHQL_UNAUTHORIZED_ROLE: public
       HASURA_GRAPHQL_LOG_LEVEL: debug
       HASURA_GRAPHQL_ENABLE_CONSOLE: 'true'
+      HASURA_GRAPHQL_EXPERIMENTAL_FEATURES: ${HASURA_GRAPHQL_EXPERIMENTAL_FEATURES}
```
Could you please fall back to some defaults here? This should probably be added to the example docker compose file as well.

szilarddoro reviewed (changes requested):
Could you please revert the PNPM lock file to use the older syntax? We have not yet bumped the PNPM version ourselves in this repo. We should do that separately.

I'll look into this deeper a bit later. Hopefully we can figure out something for the failing test you mentioned.

joshuadutton commented:
Thanks for the feedback. I'll make the changes and update the PR tomorrow. FYI, I'm in MST timezone.

joshuadutton commented:
It's taking me longer than expected because some enum types were added, and that has cascading effects to some of the input types. I've addressed all the feedback and I've also pushed a WIP commit with some transforms created and the rest stubbed out. I'll finish the rest soon.

xmlking commented:
waiting for this feature :(

dbarrosop commented:
With the migration to go this is no longer needed as we are talking to postgres directly.

Thanks a lot for setting this in motion though.

xmlking commented:
after enabling  hasura `naming convention`, auth server is not starting. is there any workaround ?

```
      HASURA_GRAPHQL_EXPERIMENTAL_FEATURES: naming_convention
      HASURA_GRAPHQL_DEFAULT_NAMING_CONVENTION: graphql-default
```

```log
auth  | {"level":"info","message":"Log level: info"}
auth  | {"level":"info","message":"Waiting for Hasura to be ready..."}
auth  | {"level":"info","message":"Hasura is ready"}
auth  | {"level":"info","message":"Applying SQL migrations..."}
auth  | {"level":"info","message":"SQL migrations applied"}
auth  | {"level":"info","message":"Applying metadata..."}
auth  | {"level":"info","message":"Metadata applied"}
auth  | /nix/store/pzr5969fyqqii70qgmfnmfdi499s12az-node_modules-prod/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:340
auth  |                         throw new types_1.ClientError(__assign(__assign({}, errorResult), { status: response.status, headers: response.headers }), { query: query, variables: variables });
auth  |                               ^
auth  |
auth  | ClientError: 'insertAuthRoles' has no argument named 'on_conflict': {"response":{"errors":[{"message":"'insertAuthRoles' has no argument named 'on_conflict'","extensions":{"path":"$.selectionSet.insertAuthRoles","code":"validation-failed"}}],"status":200,"headers":{}},"request":{"query":"mutation upsertRoles($roles: [authRoles_insert_input!]!) {\n  insertAuthRoles(\n    objects: $roles\n    on_conflict: {constraint: roles_pkey, update_columns: []}\n  ) {\n    affected_rows\n    returning {\n      role\n    }\n  }\n}","variables":{"roles":[{"role":"me"},{"role":"user"}]}}}
auth  |     at /nix/store/pzr5969fyqqii70qgmfnmfdi499s12az-node_modules-prod/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:340:31
auth  |     at step (/nix/store/pzr5969fyqqii70qgmfnmfdi499s12az-node_modules-prod/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:63:23)
auth  |     at Object.next (/nix/store/pzr5969fyqqii70qgmfnmfdi499s12az-node_modules-prod/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:44:53)
auth  |     at fulfilled (/nix/store/pzr5969fyqqii70qgmfnmfdi499s12az-node_modules-prod/node_modules/.pnpm/graphql-request@3.7.0_graphql@16.2.0/node_modules/graphql-request/dist/index.js:35:58)
auth  |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
auth  |   response: {
auth  |     errors: [
auth  |       {
auth  |         message: "'insertAuthRoles' has no argument named 'on_conflict'",
auth  |         extensions: {
auth  |           path: '$.selectionSet.insertAuthRoles',
auth  |           code: 'validation-failed'
auth  |         }
auth  |       }
auth  |     ],
auth  |     status: 200,
auth  |     headers: Headers {
auth  |       [Symbol(map)]: [Object: null prototype] {
auth  |         date: [ 'Thu, 02 May 2024 22:39:35 GMT' ],
auth  |         'content-type': [ 'application/json; charset=utf-8' ],
auth  |         'content-length': [ '162' ]
auth  |       }
auth  |     }
auth  |   },
auth  |   request: {
auth  |     query: 'mutation upsertRoles($roles: [authRoles_insert_input!]!) {\n' +
auth  |       '  insertAuthRoles(\n' +
auth  |       '    objects: $roles\n' +
auth  |       '    on_conflict: {constraint: roles_pkey, update_columns: []}\n' +
auth  |       '  ) {\n' +
auth  |       '    affected_rows\n' +
auth  |       '    returning {\n' +
auth  |       '      role\n' +
auth  |       '    }\n' +
auth  |       '  }\n' +
auth  |       '}',
auth  |     variables: { roles: [ { role: 'me' }, { role: 'user' } ] }
auth  |   }
auth  | }
auth  |
auth  | Node.js v18.19.1
auth  | {"time":"2024-05-02T22:39:36.162444086Z","level":"ERROR","msg":"node server failed","error":"exit status 1"}
auth  | {"time":"2024-05-02T22:39:36.162501336Z","level":"INFO","msg":"shutting down server"}
auth  | {"time":"2024-05-02T22:39:36.162541044Z","level":"ERROR","msg":"server failed","error":"http: Server closed"}
auth  | {"time":"2024-05-02T22:40:36.290321169Z","level":"INFO","msg":"auth v0.29.3"}
```

dbarrosop commented:
you will need to wait until the migration to go is complete.
